File: src/main/java/com/talkdesk/todolist/repository/TodoListItemRepository.java
package com.talkdesk.todolist.repository;

import com.talkdesk.todolist.model.TodoListItem;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;

public interface TodoListItemRepository extends ReactiveCrudRepository<TodoListItem, String> {

    public Flux<TodoListItem> findAllByUserId(String userId);

}

File: src/main/kotlin/com/talkdesk/todolist/repository/TodoListItemRepository.kt
package com.talkdesk.todolist.repository

import com.talkdesk.todolist.model.TodoListItem
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux

interface TodoListItemRepository : ReactiveCrudRepository<TodoListItem, String> {

    fun findAllByUserId(userId: String): Flux<TodoListItem>

}
